variables.tf

```diff
@@ -308,3 +308,225 @@ variable "check_ec2_imdsv2" {
+
+variable "enable_efs_encrypted_check" {
+  description = "Enable efs-encrypted-check rule"
+  type        = bool
+  default     = true
+}
+
+variable "kms_key_id" {
+  description = "Amazon Resource Name (ARN) of the KMS key that is used to encrypt the EFS file system."
+  type        = string
+  default     = "keys,csv"
+}
+
+variable "check_elb_deletion_protection_enabled" {
+  description = "Enable elb-deletion-protection-enabled rule"
+  type        = bool
+  default     = true
+}
+
+variable "check_elb_logging_enabled" {
+  description = "Enable elb-logging-enabled rule"
+  type        = bool
+  default     = false
+}
+
+variable "elb_logging_s3_buckets" {
+  description = "Comma-separated list of Amazon S3 bucket names for Amazon ELB to deliver the log files."
+  type        = string
+  default     = ""
+}
+
+variable "check_iam_policy_no_statements_with_admin_access" {
+  description = "Enable iam-policy-no-statements-with-admin-access rule"
+  type        = bool
+  default     = true
+}
+
+variable "check_iam_policy_no_statements_with_full_access" {
+  description = "Enable iam-policy-no-statements-with-full-access rule"
+  type        = bool
+  default     = true
+}
+
+variable "exclude_permission_boundary" {
+  description = "Boolean to exclude the evaluation of IAM policies used as permissions boundaries. If set to 'true', the rule will not include permissions boundaries in the evaluation. Otherwise, all IAM policies in scope are evaluated when set to 'false.'"
+  type        = bool
+  default     = false
+}
+
+variable "check_nacl_no_unrestricted_ssh_rdp" {
+  description = "Enable nacl-no-unrestricted-ssh-rdp rule"
+  type        = bool
+  default     = true
+}
+
+variable "check_internet_gateway_authorized_vpc_only" {
+  description = "Enable internet-gateway-authorized-vpc-only rule"
+  type        = bool
+  default     = false
+}
+
+variable "authorized_vpc_ids" {
+  description = "Comma-separated list of the authorized VPC IDs with attached IGWs. If parameter is not provided all attached IGWs will be NON_COMPLIANT."
+  type        = string
+  default     = "keys,csv"
+}
+
+variable "check_rds_snapshot_encrypted" {
+  description = "Enable rds-snapshot-encrypted rule"
+  type        = bool
+  default     = true
+}
+
+variable "check_rds_cluster_deletion_protection_enabled" {
+  description = "Enable rds-cluster-deletion-protection-enabled rule"
+  type        = bool
+  default     = true
+}
+
+variable "check_db_instance_backup_enabled" {
+  description = "Enable db-instance-backup-enabled rule"
+  type        = bool
+  default     = false
+}
+
+variable "check_s3_bucket_level_public_access_prohibited" {
+  description = "Enable s3-bucket-level-public-access-prohibited rule"
+  type        = bool
+  default     = false
+}
+
+variable "s3_bucket_public_access_prohibited_exclusion" {
+  description = "Comma-separated list of known allowed public Amazon S3 bucket names."
+  type        = string
+  default     = "bucket_names,csv"
```

Comment:
What does `bucket_names,csv` mean here?

Reply:
I was using these as temporary default values. Terraform comes back with errors if these values are left blank by default even if the rule is not enabled. If you have any suggestions, please do let me know.

Reply:
What are the errors?

Reply:
example error: 
`│ Error: Error creating AWSConfig rule: Failed to create AWSConfig rule: InvalidParameterValueException: Blank spaces are not acceptable for input parameter: KmsKeyId.`
tf version >= 0.12.7
Provider >= 2.70

Reply:
Could be wrong here, but if there is default value on this variable that implies the variable is optional. But if TF errs when the variable default is left blank, then that should be a sign that this variable is required and should not have a default value. Rather the user should always provide a value. Is there a way to implement this?

Reply:
Seems logical Minh. Can we test it?

Reply:
@minhlai @jsarnovsky What are next steps?